buildbot/osuosl/master/master.cfg

```diff
@@ -61,9 +61,18 @@ c['collapseRequests'] = buildrequest.collapseRequests
 
 from buildbot.plugins import util
 
-c['builders'] = builders = [
-  util.BuilderConfig(**b) for b in config.builders.all
-]
+def nextSVEBuild(builder, reqs):
+    if bool(builder.building):
+        return False
+    return min(reqs, key=lambda req: req.submittedAt)
+
+builders = []
+for b in config.builders.all:
+    nextBuild = None
+    if b['name'].startswith("clang-aarch64-sve-"):
+        nextBuild = nextSVEBuild
+    builders.append(util.BuilderConfig(**b, nextBuild=nextBuild))
+c['builders'] = builders
```

Comment:
Thanks for doing this, I know it's a pain to setup a whole local infra for these things.

I get the logic here but I think we'll want it to be more general:
* Add an attribute to our builder class `max_simultaneous_builds` or `single_build_only` if we're being binary about it.
* Somehow get to that attribute here and check for it instead of name matching.

Also I was thinking this could be done to accept a number of builds, but I don't know if we have that information in the nextbuild callback. For Linaro's specific purpose right now, checking for 0 or 1 is fine, but as a general feature, "start a new build if active builds is < N" is better.

Reply:
Thanks for the review. The last commit adds `max_simultaneous_builds` to linaro-g3-* builders and use it to set `nextBuild`.
It seemed better to me to avoid changing BuilderConfig class, that is in another repository, so I added `max_simultaneous_builds` to the dictionaries that are used to create BuilderConfig instances.